app/api/profiles/[id]/reject/route.ts

```diff
@@ -0,0 +1,35 @@
+import { saveRejectingReason } from '@/backend/profile/rejection.service'
+import { findUserByEmail } from '@/backend/user/user.service'
+import { authOptions } from '@/lib/auth'
+import { Role } from '@prisma/client'
+import { getServerSession } from 'next-auth'
+import { NextRequest, NextResponse } from 'next/server'
+
+export async function POST(
+  request: NextRequest,
+  { params }: { params: { id: string } },
+) {
+  const profileId = params.id
+
+  const session = await getServerSession(authOptions)
+  const user = session && (await findUserByEmail(session.user.email))
+
+  if (!user || !user?.roles.includes(Role.MODERATOR)) {
```

Comment:
**nitpick:** As this check is gonna be used frequently, we can move it to some `utils/auth` function, like:

```ts
requireUserRoles([Role.Moderator])
```